File: solutions/c/darts/1/darts.c
#include "darts.h"
#include "math.h"

int score(coordinate_t landing_position)
{
    float x = landing_position.x;
    float y = landing_position.y;
    float r=sqrt((x*x)+(y*y));
    if((r<=1) && (r>=0))
        return 10;
    else if (r<=5)
        return 5;
    else if (r<=10)
        return 1;
    return 0;
}

File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    int score(double xOfDart, double yOfDart) {
        double distance = Math.hypot(xOfDart, yOfDart);
        final double OUTER_CIRCLE = 10;
        final double MIDDLE_CIRCLE = 5;
        final double INNER_CIRCLE = 1;
        
        if (distance > OUTER_CIRCLE) {
            return 0;
        }
        if (distance > MIDDLE_CIRCLE) {
            return 1;
        }
        if (distance > INNER_CIRCLE) {
            return 5;
        }
        return 10;
    }
}
